src/lib/ngsi/SubscribeError.cpp

```diff
@@ -46,37 +47,14 @@ SubscribeError::SubscribeError()
 *
 * SubscribeError::toJson -
 */
-std::string SubscribeError::toJson(RequestType requestType, bool comma)
+std::string SubscribeError::toJson(void)
 {
-  std::string  out;
+  JsonHelper jh;
 
-  out += JSON_VALUE("error", errorCode.reasonPhrase.c_str());
-  out += ",";
-  out += JSON_VALUE("description", errorCode.details.c_str());
+  jh.addString("error", errorCode.reasonPhrase);
+  jh.addString("description", errorCode.details);
 
-
-  if (requestType == UpdateContextSubscription)
-  {
-    //
-    // NOTE: the subscriptionId must have come from the request.
-    //       If the field is empty, we are in unit tests and I here set it to all zeroes
-    //
-    if (subscriptionId.get() == "")
-    {
-      subscriptionId.set("000000000000000000000000");
-    }
-    out += ",";
-    out += JSON_PROP("affectedItems") + "[" + JSON_STR(subscriptionId.toJson(requestType, true)) + "]";
-  }
-  else if ((requestType          == SubscribeContext)           &&
-           (subscriptionId.get() != "000000000000000000000000") &&
-           (subscriptionId.get() != ""))
-  {
-    out += ",";
-    out += JSON_PROP("affectedItems") + "[" + JSON_STR(subscriptionId.toJson(requestType, true)) + "]";
-  }
-
-  return out;
+  return jh.str();
```

Comment:
This we can do as we have deprecated API V1, I imagine ...
But, even deprecated, the broker still supports the requests, right?

Reply:
I think so. What I think is happening is that NGSIv2 never renders this object. NGSIv1 still rendering it using the toJsonV1() method.

NTC (I guess)

Reply:
If still used in NGSIv1, with this change, the NGSIv1 protocol is no longer followed.
That was my question, actually.
Not that I am very fond of NGSIv1 but, it is defined as it is ... with sub-id as mandatory field.

Or, am I wrong?

Reply:
NGSIv1 protocol is no longer followed... in a place where doesn't need to follow. Note that here we are inside toJson() method that is only used by NGSIv2 logic.

NGSIv1 render logic is done in toJsonV1() that, as far as I understand, it still using this stuff.

(Maybe I haven't understood the question and my answer doesn't make sense :)

Reply:
I think I understand now ... Finally!
It's a bad diff. The content that was for NGSIv1 seems removed but it has probably simply been moved to a toJsonV1 method.
NTC
NTC